### ISSUE TITLE
False positive: "someParameter" property must not have unevaluated properties

### ISSUE THREAD

**Issue by ostrolucky** [stoplightio/spectral#2739]
**Describe the bug**
Getting a weird error on schemas generated by [API Platform](https://api-platform.com/). This is same as reported at https://github.com/APIDevTools/swagger-cli/issues/83 with addition of `allowEmptyValue`. Without it, it doesn't reproduce.

**To Reproduce**

Run `spectral lint` with `spectral:oas` ruleset on following schema
```yaml
openapi: 3.1.0
info:
    title: Test
    version: 1.0.0

servers:
    - url: "https://example.com/{someParameter}"
      variables:
          someParameter:
              default: sometest123
              description: Some description!
              allowEmptyValue: false

paths: {}

```

will get you
```
 9:25    error  oas3-schema            "someParameter" property must not have unevaluated properties.  servers[0].variables.someParamete
```

**Expected behavior**
No errors

**Environment (remove any that are not applicable):**
 - Library version: 6.14.1
 - OS: MacOS 15.1

**Additional context**

Apparently, this issue doesn't happen if you either change version from `3.1.0` to `3.0.0`, or remove `allowEmptyValue`

**Comment by dodabyte** [comment]
I have a similar problem.

**Comment by juan-carvajal** [comment]
Same problem here:
```yaml
        - name: parent-tag
          in: header
          description: Version string of parent model, if any model
          required: false
          allowEmptyValue: true
          schema:
            type: string
            example: parentTag
```
